### PR TITLE
Decouple pushnotifications

### LINKED ISSUE #1009: Move business logic of notifications out of models
In GitLab by @se-bastiaan on Feb 23, 2020, 16:30

### One-sentence description

Move business logic of notifications out of models

### Why?
Clean up the models

### Current implementation

Code in the models (mostly events)

### Suggested implementation

Use signals

### LINKED ISSUE #2755: Remove test dependencies from push notifications on events
### Describe the change
Remove test dependencies from push notifications on events

### Motivation
Better modularity of apps

### Current implementation
The tests of the app depend on events

### LINKED ISSUE #2758: Make push notifications depend on newsletters, events, photos and pizzas app instead of other way around
### Describe the change
Make push notifications depend on newsletters, events, photos and pizzas app instead of other way around

### Motivation
Better modularity of apps

### Current implementation
A lot of apps depend on pushnotifications

### Suggested implementation
Use signals and make the push notifications app depend on the other apps.

### Additional context
This makes it easy to simply deactivate push notifications without breaking anything else

### PR DESCRIPTION
**Pull request by DeD1rk**
Closes #2758.
Closes #1009. 
Closes #2755.

### Summary
This removes the dependencies from various apps on pushnotifications, by scheduling/sending notifications from signals instead of within models/views themselves.

### How to test
Steps to test the changes you made (this is kinda already done by the tests I wrote):
1. Try out things that should trigger a pushnotification.
2. Receive pushnotifications (or check that Message/ScheduledMessage objects are created).

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -2,3 +2,7 @@ version: "2"
 exclude_patterns:
   - "**/migrations/*.py"
   - "docs/"
+checks:
+  complex-logic:
+      config:
+        threshold: 15
```
Comment (pingiun) [inline .codeclimate.yml:8]: lol nice

**Comment by pingiun** [comment]
Do we really need 5 different signals?

**Comment by DeD1rk** [comment]
> Do we really need 5 different signals?

Yeah, two of them could be combined but that would hurt code quality

**Review by JobDoesburg — APPROVED** [review]
Love this

**Pull request merged**